common/etc/nginx/include/s3gateway.js

```diff
@@ -431,6 +458,10 @@ function signatureV2(r, bucket, credentials) {
      * nginx, then in S3 we need to request /?delimiter=/&prefix=dir1/
      * Thus, we can't put the path /dir1/ in the string to sign. */
     var uri = _isDirectory(r.variables.uri_path) ? '/' : r.variables.uri_path;
+    // For static website we want the path + index.html
+    if (static_hosting && _isDirectory(r.variables.uri_path)){
+        uri = r.variables.uri_path + "index.html"
```

Comment:
Please put the string literal `index.html` in a top level variable and reference it.

Reply:
I've merged @Victrid 's PR with mine.  Still have some cleanup and documentation, will get it in in the next few days.

Reply:
@dekobon I think all the changes you requested have been made.